students/Kristy_Martini/lesson04/mailroom.py

```diff
@@ -0,0 +1,139 @@
+        if name_to_thank == "quit":
+            prompt_user(newReport)
+    for donor in newReport.donors:
+        if donor.name == name_to_thank:
+            donor_found = True
+            donation_amount = input("Please enter the donation amount for which you want to thank this donor: ")
+            if donation_amount == "quit":
+                prompt_user(newReport)
+            donor.add_gift(int(donation_amount))
+            send_thank_you(donor)
+            prompt_user(newReport)
+    if donor_found == False:
+        print(name_to_thank, " is a new donor. They will be added to our database.")
+        donation_amount = input("Please enter the donation amount for which you want to thank this donor: ")
+        if donation_amount == "quit":
+            prompt_user(newReport)
+        newDonor = Donor(name_to_thank, int(donation_amount), 1, int(donation_amount))
+        newReport.add_donor(newDonor)
+        send_thank_you(donor)
+        prompt_user(newReport)
+
+def send_thank_you(donor):
+    """ Print a thank you to the donor"""
+    output_dir = os.getcwd()
+    output_file = donor.name + ".txt"
+    with open(os.path.join(output_dir, output_file), 'w') as f:
+        f.write("Thank you " + donor.name + " for your charitable gift to our organization.\n We could not operate without the generostiy of donors like yourself.")
+        f.write("Your generous gifts swill allow us to continue to serve our community in the hopes of a better world")
+
+def send_thank_you_multiple(newReport):
+    for donor in newReport.donors:
+        send_thank_you(donor)
+    prompt_user(newReport)
+
+def prompt_user(newReport=None):
+    if newReport is None:
+        newReport = Report()
+
+    """ Displays menu of user options"""
+    arg_dict = {"1": check_name, 
+                "2": newReport.create_report, 
+                "3": send_thank_you_multiple, 
+                "4": quit}
+
+    print("Hello! Welcome to the donation portal. You may enter 'quit' any time you are prompted to return to this screen.")
+    print("What would you like to do today?")
+    print("1. Send a Thank You to a single donor.")
+    print("2. Create a Report.")
+    print("3. Send letters to all donors.")
+    print("4. Quit")
+    choice = input("Please enter the number associated with your choice: ")
+
+    arg_dict[choice](newReport)
+    
+    # if choice == "1":
+    #     if newReport is None:
+    #         newReport = Report()
+    #     check_name(newReport)
+    # elif choice == "2":
+    #     if newReport is None:
+    #         newReport = Report()
+    #     newReport.create_report()
+    #     prompt_user(newReport)
+    # elif choice == "3":
+    #     if newReport is None:
+    #         newReport = Report()
+    #     send_thank_you_multiple(newReport)
+    # elif choice == "4":
+    #     quit()
+
+if __name__ == "__main__":
+    prompt_user()
```

Comment:
in this revision of the mailroom you're expected to use dictionary for holding donors data - please update your code with dictionaries.